### PR TITLE
Ko 247

### LINKED ISSUE #247: implement java submissions
## Summary

- right now submissions work with python, we need to have them work with java also.

### Basic example

- a user submitting a java file should result in a successful submission

### Motivation

- to allow users to submit java code

### PR DESCRIPTION
**Pull request by MrKevinOConnell**
# Description

Fixes #247 

## Type of change

- [x] 💡 New feature

# How Has This Been Tested?

- tested with a java submission that was correct
- with a java submission that isn't correct
- tested with a java submission that doesn't compile
- [x] Tested individually.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] My changes do not break any features.

### CODE REVIEW

File: backend/src/api/submissions/postSubmissions.ts [hunk 1]
```diff
@@ -184,115 +184,117 @@ export const postSubmissions = async (req: Request, res: Response): Promise<void
...
-        }
-
-        submission.status = status
-        // Calculate Score
-        if (status == 'accepted') {
-          let minutes = 0
-          if (problem.practice) {
-            minutes = ((submission.date as any) - practice_start_date) / 60
+          )
+          console.log('res', res.data)
```
Comment (a1g3) [inline backend/src/api/submissions/postSubmissions.ts:242]: You can probably remove the console.log statements.

File: backend/src/api/submissions/postSubmissions.ts [hunk 2]
```diff
@@ -184,115 +184,117 @@ export const postSubmissions = async (req: Request, res: Response): Promise<void
...
+          console.log('test out', test.out)
+          console.log('date output', res.data.run.output)
+          console.log('is same', (res.data.run.output as string) == (test.out as string))
+          test['stdout'] = res.data.run.code == 0 ? res.data.run.stdout : res.data.run.stderr
+          if ((res.data.run.output.trim() as string) == (test.out.trim() as string) || res.data.run.code !== 0) {
```
Comment (a1g3) [inline backend/src/api/submissions/postSubmissions.ts:247]: For this, wouldn't we want to reject the submission of the process returns a non-zero code?

File: frontend/src/pages/blue/Submit.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ const Submit = (): JSX.Element => {
     const formData = new FormData()
     formData.set('pid', problem.pid)
     formData.set('source', file, file.name)
-    formData.set('language', language.key)
+    formData.set('language', language.value)
```
Comment (a1g3) [inline frontend/src/pages/blue/Submit.tsx:68]: Was this the wrong thing from before?

**Review by a1g3 — CHANGES_REQUESTED** [review]

File: backend/src/api/submissions/postSubmissions.ts (hunk 1 above)
Comment (MrKevinOConnell) [reply backend/src/api/submissions/postSubmissions.ts:242]: 👍🏻

File: backend/src/api/submissions/postSubmissions.ts (hunk 2 above)
Comment (MrKevinOConnell) [reply backend/src/api/submissions/postSubmissions.ts:247]: yea your right, good call

**Review by a1g3 — APPROVED** [review]

**Pull request merged**